src/com/itschool/session6/homework/challenge15/Challenge15.java

```diff
@@ -0,0 +1,20 @@
+package com.itschool.session6.homework.challenge15;
+
+public class Challenge15 {
+    public static void main(String[] args) {
+        authDemo();
+    }
+    static void authDemo() {
+        boolean hasUsername = true;
+        boolean hasPassword = false;
+
+        //check authentication conditions
+        if (hasPassword && hasUsername) {
+            System.out.println("Authentication successfull");
+        } else if (hasUsername && !hasPassword) {
```

Comment:
`!hasPassword` is always true here because otherwise the code would have gone to the above if condition. However, I think you could refactor it to something like (also using dynamic variables captured from a `Scanner`):
```
Scanner scanner = new Scanner(System.in);
boolean hasUsername = scanner.nextBoolean();
boolean hasPassword = scanner.nextBoolean();

//check authentication conditions
if (hasUsername && hasPassword) {
	System.out.println("Authentication successful");
} else {
	if (hasUsername) {
		System.out.println("Password is incorrect.");
	} else {
		System.out.println("Authentication failed.");
	}
}
```

Reply:
solved